conda_concourse_ci/execute.py

```diff
@@ -96,6 +96,7 @@ def collect_tasks(path, folders, matrix_base_dir, channels=None, steps=0, test=F
         # each platform will be submitted with a different label
         for platform in platforms:
             index_key = '-'.join([platform['platform'], str(platform['arch'])])
+            config.variant = platform.get('variants') or {}
```

Comment:
there's an important distinction between ``variant`` and ``variants``.  The former has only one value per key, whereas the latter may have more than one per key, and undergoes the whole matrix stuff.  We need to allow that latter behavior here.  That's what I was talking about in https://github.com/ContinuumIO/automated-build/pull/8#issuecomment-435970310 with the reference to https://github.com/conda/conda-build/blob/master/conda_build/cli/main_render.py#L179

I believe that you will want to store the return value of that function as the value of config.variants here.  The rendering later in the process will take care of the matrix stuff, and you'll be left with a bunch of builds, each with only a single variant (one combination of settings from the matrix).

Reply:
One other gotcha to pay attention to:

On the way in, ``variants`` is a dictionary with keys having potentially multiple values (referred to as a "dict of lists").  After the variants are all merged, ``variants`` is a list of dicts, and each dict is a variant.  This is generally the form you'll find ``config.variants`` in.

This function: https://github.com/conda/conda-build/blob/master/conda_build/variants.py#L478

takes in the "dict of lists" and returns the "list of dicts" that you should store as the value of ``config.variants``